cygrid/tests/test_cygrid.py

```diff
@@ -58,26 +62,132 @@ def setup(self):
             'CRVAL2': mapcenter[1],
             }
 
-    def teardown(self):
-
-        pass
-
-    def test_gridding(self):
-
         kernelsize_fwhm = self.beamsize_fwhm / 2
         kernelsize_sigma = kernelsize_fwhm / 2.35
         support_radius = 3. * kernelsize_sigma
         hpx_min_res = kernelsize_sigma / 2.
 
-        mygridder = WcsGrid(self.header)
-        mygridder.set_kernel(
+        self.kernel_args = (
             'gauss1d',
             (0.5 / kernelsize_sigma ** 2,),
             support_radius,
             hpx_min_res,
             )
 
-        mygridder.grid(self.xcoords, self.ycoords, self.signal[:, np.newaxis])
+        self.test_maps = np.load(get_pkg_data_filename(
+            'data/cygrid_test_maps.npy'
+            ))
+
+    def teardown(self):
+
+        pass
+
+    def _do_gridding2d(self, signal, header, **kwargs):
+
+        mygridder = cygrid.WcsGrid(header, **kwargs)
+        mygridder.set_kernel(*self.kernel_args)
+
+        mygridder.grid(self.xcoords, self.ycoords, signal[:, np.newaxis])
+
+        assert_allclose(
+            mygridder.get_datacube()[0],
+            self.test_maps[0],
+            atol=1.e-6
+            )
+
+        # test automatic patching of input signal array size (to 2D)
+        mygridder.grid(self.xcoords, self.ycoords, signal)
+
+        assert_allclose(
+            mygridder.get_datacube()[0],
+            self.test_maps[0],
+            atol=1.e-6
+            )
+
+    def _do_gridding3d(self, signal, header, **kwargs):
+
+        mygridder = cygrid.WcsGrid(header, **kwargs)
+        mygridder.set_kernel(*self.kernel_args)
+
+        mygridder.grid(self.xcoords, self.ycoords, signal)
+
+        if kwargs.get('do_store', False):
+            np.save('/tmp/cygrid_test_maps.npy', mygridder.get_datacube())
+
+        assert_allclose(mygridder.get_datacube(), self.test_maps, atol=1.e-6)
+
+    def test_gridding2d_naxis3_implicit(self):
+        '''
+        Implicitly set 3rd (redundant) axis in header
+        '''
+
+        header = self.header.copy()
+        header['NAXIS'] = 3
+        header['NAXIS3'] = 1
+        self._do_gridding2d(self.signal, header)
+
+    def test_gridding2d_naxis3_explicit(self):
+        '''
+        Explicitly set 3rd via kwargs
+        '''
+
+        self._do_gridding2d(self.signal, self.header, naxis3=1)
+
+    def test_gridding2d_naxis3_auto(self):
+        '''
+        Automatic handling of naxis3 (aka set to One, if not present)
```

Comment:
Okay, fully understood it now! Took me a while, haven't had enough coffee yet...